client/src/routes/login/tabs/SignIn.tsx

```diff
@@ -30,6 +46,35 @@ function SignIn() {
 
 	const signIn = (e: React.FormEvent<HTMLFormElement>) => {
 		e.preventDefault();
+
+		(async () => {
+
+			const params = buildRequestParams({
+				email: email,
+				password: password,
+				persistentLogin: rememberMe,
+			});
+
+			const request = await fetch(`api/signin?${params}`, {
+			  method: 'POST'
+			});
+
+			const response = await request.json();
+		  
+			if(response != 'Invalid Credentials') {
+				
+				user.setUser({
+					email: email,
+					persistentLogin: rememberMe,
+					token: response
```

Comment:
the context is removed when the user closes the page or when he refreshes the page, add the token to the localstorage so the token can be used the next time the user comes to the site you check the token.
that the only thing missing.